x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -381,7 +384,12 @@ public IndicesAccessControl authorize(
             totalResourceCount += resource.size();
         }
 
-        final boolean overallGranted = isActionGranted(action, resources);
+        final boolean overallGranted;
+        if (parentAuthorization.isPresent() && parentAuthorization.get().granted()) {
+            overallGranted = true;
+        } else {
+            overallGranted = isActionGranted(action, resources);
+        }
```

Comment:
My impression is that the `IndicesPermission` is not the right abstraction level to act on the pre-authorization. I would think that somewhere in the `RBACEngine` would be better, so that it completely bypasses `role#authorize`, like the child actions authz currently does.

I realize that the access control still needs to be computed the regular way, but I'm not a fan of this separated approach either. My preference would be to take the pre-authz path only if the role doesn't have DLS/FLS perms.

Maybe I should explain in more details in sync?